blockchain/blockdao/blockdao.go

```diff
@@ -327,8 +325,6 @@ func (dao *blockDAO) GetReceipts(blkHeight uint64) ([]*action.Receipt, error) {
 
 func (dao *blockDAO) PutBlock(blk *block.Block) error {
 	err := func() error {
-		dao.mutex.Lock()
-		defer dao.mutex.Unlock()
 		if err := dao.putBlock(blk); err != nil {
```

Comment:
i think our agreed coding practice is public interface should use lock, and private not
so we should keep this, see comments in FileIndex() below